isic_archive/__init__.py

```diff
@@ -74,62 +75,66 @@ def clearRouteDocs():
         routes[routeResource][routePath][routeMethod] = routeOperation
 
 
-def load(info):
-    sentry_sdk.init(environment=os.getenv('SENTRY_ENVIRONMENT'))
-    # set the title of the HTML pages
-    info['serverRoot'].updateHtmlVars({'title': 'ISIC Archive'})
-
-    # add event listeners
-    events.bind('rest.get.describe/:resource.after',
-                'onDescribeResource', onDescribeResource)
-
-    # add custom model searching
-    resource.allowedSearchTypes.update({
-        'image.isic_archive',
-        'study.isic_archive',
-    })
-
-    # register licenses for template usage
-    mail_utils.addTemplateDirectory(
-        os.path.join(info['pluginRootDir'], 'server', 'license_templates'),
-        prepend=True)
-
-    # add static file serving
-    info['config']['/uda'] = {
-        'tools.staticdir.on': 'True',
-        'tools.staticdir.dir': os.path.join(info['pluginRootDir'], 'custom')
-    }
-
-    # add dynamic root routes
-    # root endpoints -> where a user may go and expect a UI
-    class Root(object):
-        pass
-    legacyWebroot = Root()
-    legacyWebroot.segment = staticFile(
-        os.path.join(info['pluginRootDir'], 'custom', 'phase1.html'))
-    registerPluginWebroot(legacyWebroot, 'markup')
-
-    # create all necessary users, groups, collections, etc
-    provisionDatabase()
-
-    # add api routes
-    # remove docs for default Girder API, to simplify page
-    clearRouteDocs()
-
-    # Customize API docs template
-    baseTemplateFilename = info['apiRoot'].templateFilename
-    info['apiRoot'].updateHtmlVars({
-        'baseTemplateFilename': baseTemplateFilename
-    })
-    templatePath = os.path.join(info['pluginRootDir'], 'server', 'isic_api_docs.mako')
-    info['apiRoot'].setTemplatePath(templatePath)
-
-    # TODO: nest these under a "/isic" path?
-    info['apiRoot'].annotation = api.AnnotationResource()
-    info['apiRoot'].dataset = api.DatasetResource()
-    info['apiRoot'].image = api.ImageResource()
-    info['apiRoot'].redirects = api.RedirectsResource()
-    info['apiRoot'].segmentation = api.SegmentationResource()
-    info['apiRoot'].study = api.StudyResource()
-    info['apiRoot'].task = api.TaskResource()
-    api.attachUserApi(info['apiRoot'].user)
+class IsicArchive(GirderPlugin):
+    DISPLAY_NAME = 'ISIC Archive'
+
+    def load(self, info):
+        sentry_sdk.init(environment=os.getenv('SENTRY_ENVIRONMENT'))
+
+        getPlugin('oauth').load(info)
+        getPlugin('large_image').load(info)
+
+        # set the title of the HTML pages
+        info['serverRoot'].updateHtmlVars({'title': 'ISIC Archive'})
```

Comment:
@brianhelba Why is this still here? Shouldn't `brandName` get set via a setting?